reference-implementation/to-upstream-wpts/writable-streams/general.js

```diff
@@ -80,3 +80,22 @@ test(() => {
     }
   );
 }, 'ws.getWriter() on an errored WritableStream');
+
+promise_test(() => {
+  const ws = new WritableStream({});
+
+  const writer = ws.getWriter();
+  writer.releaseLock();
+
+  return writer.closed.then(
```

Comment:
Maybe use the weird promise_rejects() function from testharness.js? Something like

return promise_rejects(t, 'TypeError', writer.closed, 'closed promise rejects')
    .then(promise_rejects(t, 'TypeError', writer.ready, 'ready promise rejects');